freeipa-client/src/main/java/com/sequenceiq/freeipa/client/FreeIpaClient.java

```diff
@@ -284,6 +289,13 @@ public int getUsernameLength() throws FreeIpaClientException {
         return config.getIpamaxusernamelength();
     }
 
+    public Config getConfig() throws FreeIpaClientException {
+        Map<String, Object> params = new HashMap<>();
+        params.put("all", Boolean.TRUE);
+        Config config = (Config) invoke("config_show", List.of(), params, Config.class).getResult();
```

Comment:
maybe just use ImmutableMap.of("all", Boolean.TRUE) here? also, does auto-boxing 'true' not work? (you may need to change the signature for 'invoke' to be Map<String, ? extends Object> instead of Map<String, Object>

Reply:
ImmutableMap related change I can do as separate PR since we need to change in every method call and scope for testing increases. For this PR, auto-boxing 'true' and return config can be changed. I kept xtra line for the breakpoint purpose to evaluate the values and forgot to remove that. Thanks.

Reply:
For ImmutableMap, we will revisit FreeIPAClient after this PR. For now, closing this conversation.